pkg/queue/manager_test.go

```diff
@@ -87,25 +88,107 @@ func TestAddClusterQueueOrphans(t *testing.T) {
 	ctx := context.Background()
 	manager := NewManager(kClient, nil)
+	cq := utiltesting.MakeClusterQueue("cq").Obj()
+	if err := manager.AddClusterQueue(ctx, cq); err != nil {
+		t.Fatalf("Failed adding cluster queue %s: %v", cq.Name, err)
+	}
 	for _, q := range queues {
 		if err := manager.AddLocalQueue(ctx, q); err != nil {
 			t.Fatalf("Failed adding queue %s: %v", q.Name, err)
 		}
 	}
-	cq := utiltesting.MakeClusterQueue("cq").Obj()
+
+	wantActiveWorkloads := map[string]sets.String{
+		"cq": sets.NewString("/a", "/b"),
+	}
+	if diff := cmp.Diff(wantActiveWorkloads, manager.Dump()); diff != "" {
+		t.Errorf("Unexpected active workloads after creating all objects (-want,+got):\n%s", diff)
+	}
+
+	// Recreating the ClusterQueue.
+	manager.DeleteClusterQueue(cq)
+	wantActiveWorkloads = nil
+	if diff := cmp.Diff(wantActiveWorkloads, manager.Dump()); diff != "" {
+		t.Errorf("Unexpected active workloads after deleting ClusterQueue (-want,+got):\n%s", diff)
+	}
+
 	if err := manager.AddClusterQueue(ctx, cq); err != nil {
-		t.Fatalf("Failed adding cluster queue %s: %v", cq.Name, err)
+		t.Fatalf("Could not re-add ClusterQueue: %v", err)
 	}
-	workloads := popNamesFromCQ(manager.clusterQueues[cq.Name])
+	workloads := popNamesFromCQ(manager.clusterQueues["cq"])
 	wantWorkloads := []string{"/b", "/a"}
 	if diff := cmp.Diff(wantWorkloads, workloads); diff != "" {
 		t.Errorf("Workloads popped in the wrong order from clusterQueue:\n%s", diff)
 	}
 }
 
-// TestUpdateQueue tests that workloads are transferred between clusterQueues
+// TestUpdateClusterQueue tests that a ClusterQueue transfers cohorts on update.
+func TestUpdateClusterQueue(t *testing.T) {
+	clusterQueues := []*kueue.ClusterQueue{
+		utiltesting.MakeClusterQueue("cq1").Cohort("alpha").Obj(),
+		utiltesting.MakeClusterQueue("cq2").Cohort("beta").Obj(),
+	}
+	queues := []*kueue.LocalQueue{
+		utiltesting.MakeLocalQueue("foo", defaultNamespace).ClusterQueue("cq1").Obj(),
+		utiltesting.MakeLocalQueue("bar", defaultNamespace).ClusterQueue("cq2").Obj(),
+	}
+	now := time.Now()
+	workloads := []*kueue.Workload{
+		utiltesting.MakeWorkload("a", defaultNamespace).Queue("foo").Creation(now.Add(time.Second)).Obj(),
+		utiltesting.MakeWorkload("b", defaultNamespace).Queue("bar").Creation(now).Obj(),
+	}
+	// Setup.
+	scheme := utiltesting.MustGetScheme(t)
+	ctx := context.Background()
+	cl := fake.NewClientBuilder().WithScheme(scheme).WithObjects(
+		&corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: defaultNamespace}},
+	).Build()
+	manager := NewManager(cl, nil)
+	for _, cq := range clusterQueues {
+		if err := manager.AddClusterQueue(ctx, cq); err != nil {
+			t.Fatalf("Failed adding clusterQueue %s: %v", cq.Name, err)
+		}
+	}
+	for _, q := range queues {
+		if err := manager.AddLocalQueue(ctx, q); err != nil {
+			t.Fatalf("Failed adding queue %s: %v", q.Name, err)
+		}
+	}
+	// Add inadmissible workloads.
+	for _, w := range workloads {
+		if err := cl.Create(ctx, w); err != nil {
```

Comment:
Can we use WithObjects to add them to the cache directly when building the client above?